cross-doc-explainer.md

```diff
@@ -0,0 +1,114 @@
+# Contents
+
+# Introduction
+
+See [the main explainer](explainer.md) for detailed explanation about view transitions.
+This explainer extends the feature of same-document ("SPA") transitions to also include
+cross-document transitions ("MPA").
+
+While same-document transitions are useful in themselves and an important stepping stone, allowing
+cross-document transitions would unlock a capability that so far has been available only within the
+same document.
+
+# Goals
+
+## Compatible with same-document transitions
+
+The vast majority of work for cross-document transitions is based on same-document transitions.
+This allows developers to use the same techniques for both, and to avoid big refactors if
+changing architectures.
+
+## Declarative & Customizable
+
+Cross-document transitions should work by default without JavaScript intervention, and should have
+the right JavaScript knobs for when the defaults are not enough.
+
+# How it works
+
+## In a nutshell
+Documents declare that they support same-origin cross-document transitions, using a
+[meta tag](#opt-in). When a navigation between two such documents takes place (and has no
+cross-origin redirects in the middle), the state of the old document is captured, in the same
+manner as same-origin transitions. When the new document is ready for the transition (i.e., when all render-blocking resources are ready), the new state is captured,
+and the transition continues as if it was a same-document transition, with the appropriate pseudo-elements.
+
+The new document can customize the style of the animation using the same techniques available for same-document transitions.
+
+Both documents can interrupt the transition or customize it using JavaScript in different phases.
+
+## Same-origin
+
+Cross-document view transitions are only enabled for same-origin navigations without a
+[cross-origin redirect](https://html.spec.whatwg.org/#unloading-documents:was-created-via-cross-origin-redirects).
+In the future we could examine exposing them in some way to same-site or cross-origin navigations.
+
+## Opt-in
+
+To enable cross-document transitions, both the old and new document need to add a meta tag. This tag
+indicates that the author wants to enable transitions for same-origin navigations to/from this
+Document.
+
+
+### Issues
+
+1. The meta tag can be used to opt-in to other navigation types going forward: same-document, same-site, etc.
+
+1. Using meta-tags prevents the declaration being controlled by media queries, which feels important for `prefers-reduced-motion`.
```

Comment:
There's a discussion about having opt-in be in css as well: https://github.com/w3c/csswg-drafts/issues/8048

Reply:
Added a link to that discussion